src/main/webapp/app/shared/layouts/navbar/navbar.component.ts

```diff
@@ -126,4 +141,40 @@ export class NavbarComponent implements OnInit, OnDestroy {
             }
         }
     }
+
+    loadExamIfAvailable() {
+        this.routerEventSubscription = this.router.events.pipe(filter((event: RouterEvent) => event instanceof NavigationEnd)).subscribe((event) => {
+            this.exam = undefined;
+            if (event.url.includes('exams') && !event.url.includes('management')) {
+                const routePathParams = of(event).pipe(
+                    map(() => this.route.root),
+                    map((root) => root.firstChild),
+                    switchMap((firstChild) => {
+                        if (firstChild) {
+                            return firstChild?.paramMap.pipe(map((paramMap) => [paramMap.get('courseId'), paramMap.get('examId')]));
+                        } else {
+                            return of(null);
+                        }
+                    }),
+                );
+                routePathParams.subscribe((param) => {
+                    if (param !== null) {
+                        if (param[0] !== null && param[1] !== null) {
+                            this.examParticipationService.loadExam(+param[0], +param[1]).subscribe((loadedExam) => (this.exam = loadedExam));
```

Comment:
Please do not load the exam from the server in the navbar! This is a really bad idea!!!

Also the method name is confusing, because the outer method only subscribes for router changes.
Only the inner one actually loads the exam, so I suggest to extract the inner method and rename the outer one!

I would suggest that we add some kind of examService that holds the currently visible exams (they are e.g. loaded by the courses for dashboard REST call or when students navigate into the exam. This could then notify the navbar, when an exam is available. The navbar then does not need to load it from the server!

Reply:
Refactored the code according to your suggestions 😊 
Please let me know whether the changes meet your expectations